premium/usage.go

```diff
@@ -200,6 +206,20 @@ func NewUsageClient(meta plugin.Meta, ops ...UsageClientOptions) (UsageClient, e
 			TeamNameValue: u.teamName,
 		}, nil
 	}
+	// If user wants to use the AWS Marketplace for billing, don't even try to communicate with CQ API
+	if isAWSMarketplace() {
+		cfg, err := awsConfig.LoadDefaultConfig(context.TODO())
+		if err != nil {
+			return nil, fmt.Errorf("failed to load AWS config: %w", err)
+		}
+
+		u.awsMarketPlaceClient = marketplacemetering.NewFromConfig(cfg)
+		u.teamName = "AWS_MARKETPLACE"
+		// This needs to be larger than normal, because we can only send a single usage record per second (from each compute node)
+		u.batchLimit = 1000000000
```

Comment:
I'm not sure this is necessary, but would need to verify. I think how it should work is that there is a minimum flush time that should come into effect (by default this is 10s)

Reply:
I think there is a race condition, if a batch is flushed and within the same second the sync ends then the timestamp will have the same value as timestamps are truncated to the second

Reply:
Ah, good catch... ok this should work for now but we should probably try and think of a better solution in the long term, otherwise we won't count any records if the sync fails